### DISCUSSION TITLE
Is there a way to resize nodes based on their width and height on restore?

### DISCUSSION THREAD

**Discussion by mlndshh** [xyflow/xyflow#2304] [Q&A]
I want to resize a custom node based that is resizable. I can successfully change the width and height of of a node when resizes take place. However when I save this to localstorage and restore it later, the nodes do not get restored with their width and height from earlier. 

They do still have the updated width and height attributes, but unfortunately they are not passed to the node. How exactly do I resize a node and then get it to persist when the width and height is not passed to the node? Thanks!

**Comment by moklick** [comment]
Hey @mlndshh 

this might be a bit confusing but you need to set the style to give a node dimensions:

```js
const nodeWithDimensions = {
  id: 'a',
  position: { x: 0, y: 0 },
  style: {
    width: 200,
    height: 50
  }
};
```

**Comment by mlndshh** [comment]
Thanks @moklick, does that mean I have to run setNodes after every resize?

**Reply by moklick** [reply]
Ah I got your question wrong.. When you resize a node, it gets re-measured by React Flow. The node receive the dimensions by the onNodesChange handler. You don't need to do anything.
